src/config/axios.ts

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+import {} from "../redux/app/store";
+import { url } from "../main/constants/common";
+import { logout, refreshToken } from "../redux/features/auth/authSlise";
+
+let store: any;
+
+export const setStore = (reduxStore: any) => {
+  store = reduxStore;
+};
+
+const api = axios.create({
+  baseURL: url,
+});
+
+api.interceptors.request.use((config) => {
+  if (!store) return config;
+
+  const state = store.getState();
+  const token = state.auth.accessToken;
+
```

Comment:
It has to be within the local storage if you use `persist`. It looks weird. Let`s discuss it

Reply:
@VYaniv, could you please clarify the issue? What exactly do you mean?